### PR TITLE
Fix for issue #47216

### PR DESCRIPTION
**Pull request by shivaditya-meduri**
FIX TypeError if set the weights to the current weights via `set_weights, 
Incorporated suggestions by @qlzh727 i.e. included type checking to reduce overhead.
Simply, this change will allow users to use .weights to initialize the set_weights() method of a layer without any type error improving the versatility of keras.

### CODE REVIEW

**Review by qlzh727 — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/engine/base_layer.py [hunk 1]
```diff
@@ -1868,6 +1868,10 @@ def set_weights(self, weights):
           'expecting %s weights. Provided weights: %s...' %
           (self.name, len(weights), expected_num_weights, str(weights)[:50]))
 
+    for index, item in enumerate(weights):
+      if(str(type(item))=="<class 'tensorflow.python.ops.resource_variable_ops.ResourceVariable'>"):
+          weights[index] = np.array(item)
```
[the remark this answers is not in the document]
Comment (shivaditya-meduri) [reply tensorflow/python/keras/engine/base_layer.py:1873]: Thank you for reviewing, I will make the changes!

[the remark this answers is not in the document]
Comment (shivaditya-meduri) [reply tensorflow/python/keras/engine/base_layer.py]: Hey @qlzh727, I have made the changes as per your suggestions. Can you please take a look at it?

**Review by fchollet — COMMENTED** [review]
At line 1880, `weight = weights[weight_index]`, you should do:

```
if isinstance(weight, tf_variables.Variable):
   weight = weight.numpy()
elif not isinstance(weight, np.ndarray):
   raise ValueError(...)
```

You should also add a unit test for both the new functionality and the error message.

**Comment by shivaditya-meduri** [comment]
@fchollet, Thank you for your advice, I will include the changes you suggested.

**Comment by fchollet** [comment]
Please include a unit test.

**Comment by shivaditya-meduri** [comment]
``"""
    >>> a = tf.keras.layers.Dense(1,
    ...   kernel_initializer=tf.constant_initializer(1.))
    >>> a_out = a(tf.convert_to_tensor([[1., 2., 3.]]))
    >>> a.get_weights()
    [array([[1.],
           [1.],
           [1.]], dtype=float32), array([0.], dtype=float32)]
    >>> b = tf.keras.layers.Dense(1,
    ...   kernel_initializer=tf.constant_initializer(2.))
    >>> b_out = b(tf.convert_to_tensor([[10., 20., 30.]]))
    >>> b.get_weights()
    [array([[2.],
           [2.],
           [2.]], dtype=float32), array([0.], dtype=float32)]
    >>> b.set_weights(a.weights)
    >>> b.get_weights()
    [array([[1.],
           [1.],
           [1.]], dtype=float32), array([0.], dtype=float32)]
    """

**Comment by shivaditya-meduri** [comment]
"""
    >>> a = tf.keras.layers.Dense(1,
    ...   kernel_initializer=tf.constant_initializer(1.))
    >>> a_out = a(tf.convert_to_tensor([[1., 2., 3.]]))
    >>> a.get_weights()
    [array([[1.],
           [1.],
           [1.]], dtype=float32), array([0.], dtype=float32)]
    >>> b = tf.keras.layers.Dense(1,
    ...   kernel_initializer=tf.constant_initializer(2.))
    >>> b_out = b(tf.convert_to_tensor([[10., 20., 30.]]))
    >>> b.get_weights()
    [array([[2.],
           [2.],
           [2.]], dtype=float32), array([0.], dtype=float32)]
    >>> b.set_weights(a.weights)
    >>> b.get_weights()
    [array([[1.],
           [1.],
           [1.]], dtype=float32), array([0.], dtype=float32)]
"""
import doctest
doctest.testmod()
`

**Comment by shivaditya-meduri** [comment]
@fchollet, I found this test on the tensorflow website for set_weights() method, I made modifications to the doctest to include .weights instead of get_weights() method to assign weights.

**Comment by qlzh727** [comment]
Sorry for the long wait. The Keras code base has actually been migrated to github.com/keras-team/keras. If you still would like to contribute the PR, feel free to post your change to the new repository. I am closing this PR for now.

**Pull request closed without merging**